src/ripple/basics/impl/PerfLogImp.cpp

```diff
@@ -302,10 +289,13 @@ PerfLogImp::report()
 
     Json::Value report(Json::objectValue);
     report[jss::time] = to_string(date::floor<microseconds>(present));
-    report[jss::workers] = counters_.workers_;
+    {
+        std::lock_guard lock{counters_.jobsMutex_};
```

Comment:
Nice find on this!

Reply:
It's a bit of a shame we now have to take a lock here. We could consider making `workers_` `atomic` and losing the lock here, but your call.

Reply:
Is the number of worker threads guaranteed to be the size of the jobs_ vector? Or is the jobs_ vector the list of currently executing jobs? If it's the latter, then the meaning of the jss:workers field changes. Namely, it's there to describe the total number of worker threads. That way, the user can compare the number of running jobs against this value to see if the job queue is saturated. In normal operation, workers_ stays the same after the process sets the job queue size.

Please put workers_ back the way it was. Regarding whether to make it atomic, that's probably the most sensible thing.

Reply:
The number of worker threads is guaranteed to be the size of the `jobs_` vector. Notice:

- An item in `jobs_` has job type `INVALID` and default-constructed "start time" whenever it [finishes](https://github.com/ripple/rippled/pull/3706/files#diff-64246f9206962a784344c0dbcbff598c1052de9290342f27faaafe566bfa5d1fL438) or when it is [constructed](https://github.com/ripple/rippled/pull/3706/files#diff-64246f9206962a784344c0dbcbff598c1052de9290342f27faaafe566bfa5d1fL447).
- `workers_` was assigned only when the size of `jobs_` changed during [resize](https://github.com/ripple/rippled/pull/3706/files#diff-64246f9206962a784344c0dbcbff598c1052de9290342f27faaafe566bfa5d1fL445), not when jobs started or finished.

Reporting already acquires a bunch of locks to copy out the counter for every used job type and RPC method type. I don't think the difference of one more lock is substantial enough to make it worth introducing the complexity of maintaining a second job counter.